package.json

```diff
@@ -90,7 +90,102 @@
     "pull-ws": "^3.3.2",
     "json-schema": "^0.4.0",
     "simple-get": "^4.0.1",
-    "@storybook/**/ast-types": "^0.14.2"
+    "@storybook/**/ast-types": "^0.14.2",
+    "typescript@~4.4.0": "patch:typescript@npm:4.4.4#.yarn/patches/typescript-npm-4.4.4-3fedcc07a3.patch",
+    "acorn@^7.0.0": "patch:acorn@npm:7.4.1#.yarn/patches/acorn-npm-7.4.1-f450b4646c.patch",
+    "acorn@^7.4.1": "patch:acorn@npm:7.4.1#.yarn/patches/acorn-npm-7.4.1-f450b4646c.patch",
+    "acorn@^7.1.1": "patch:acorn@npm:7.4.1#.yarn/patches/acorn-npm-7.4.1-f450b4646c.patch",
+    "acorn@7.4.1": "patch:acorn@npm:7.4.1#.yarn/patches/acorn-npm-7.4.1-f450b4646c.patch",
```

Comment:
It seems that we are effectively resolving four version specifiers for acorn to the same version (7.4.1), so can we not collapse these into: 

``` suggestion
    "acorn": "patch:acorn@npm:7.4.1#.yarn/patches/acorn-npm-7.4.1-f450b4646c.patch",
```

The same thing goes for the other indirect dependencies here.

Reply:
we cannot collapse to just 'acorn' because we also have acorn 8.8.0, 6.4.2 and 5.7.4 as transient dependencies and the patch should not be valid for those versions.

<details>
<summary><code>yarn info -A -R --dependents acorn</code> -- without any changes</summary>

```
➜ yarn info -A -R --dependents acorn 
├─ acorn@npm:5.7.4
│  ├─ Version: 5.7.4
│  │
│  ├─ Exported Binaries
│  │  └─ acorn
│  │
│  └─ Dependents
│     └─ jsdom@npm:11.5.1
│
├─ acorn@npm:6.4.2
│  ├─ Version: 6.4.2
│  │
│  ├─ Exported Binaries
│  │  └─ acorn
│  │
│  └─ Dependents
│     ├─ @gulp-sourcemaps/identity-map@npm:2.0.1
│     ├─ acorn-globals@npm:4.3.4
│     ├─ gulp-sourcemaps@npm:3.0.0
│     ├─ gulp-sourcemaps@patch:gulp-sourcemaps@npm%3A3.0.0#./.yarn/patches/gulp-sourcemaps-npm-3.0.0-1ae0fbef6d.patch::version=3.0.0&hash=352196&locator=metamask-crx%40workspace%3A.
│     ├─ webpack@npm:4.46.0
│     └─ webpack@npm:4.46.0 [15918]
│
├─ acorn@npm:7.4.1
│  ├─ Version: 7.4.1
│  │
│  └─ Exported Binaries
│     └─ acorn
│
├─ acorn@npm:8.8.0
│  ├─ Version: 8.8.0
│  │
│  ├─ Exported Binaries
│  │  └─ acorn
│  │
│  └─ Dependents
│     ├─ acorn-import-assertions@npm:1.8.0 [f6da4]
│     ├─ acorn-jsx@npm:5.3.2 [c70fa]
│     ├─ espree@npm:9.3.1
│     ├─ espree@npm:9.3.2
│     ├─ jsdom@npm:20.0.0
│     ├─ jsdom@npm:20.0.0 [15ca4]
│     ├─ terser@npm:4.8.0
│     ├─ terser@npm:5.15.0
│     ├─ ts-node@npm:10.5.0
│     ├─ ts-node@npm:10.5.0 [15918]
│     ├─ vm2@npm:3.9.11
│     ├─ webpack@npm:5.74.0
│     └─ webpack@npm:5.74.0 [37b7a]
│
└─ acorn@patch:acorn@npm%3A7.4.1#.yarn/patches/acorn-npm-7.4.1-f450b4646c.patch::version=7.4.1&hash=9c9719&locator=metamask-crx%40workspace%3A.
   ├─ Version: 7.4.1
   │
   ├─ Exported Binaries
   │  └─ acorn
   │
   └─ Dependents
      ├─ @storybook/react@npm:6.5.10
      ├─ @storybook/react@npm:6.5.10 [15918]
      ├─ acorn-globals@npm:6.0.0
      ├─ acorn-jsx@npm:5.3.2 [37b7a]
      └─ acorn-node@npm:1.8.2
```

</details>

Furthermore, as stated in description, the patching system is *very* picky. If will only apply the patch if the version number matches *exactly* what is specified this is why there are four resolutions, if i remove the one that looks the most alike (7.4.1 vs ^7.4.1) , you'll see the unpatched copy now has dependents: 

<details>
<summary><code>yarn info -A -R --dependents acorn</code> -- without ^7.4.1</summary>

```
➜ yarn info -A -R --dependents acorn
├─ acorn@npm:5.7.4
│  ├─ Version: 5.7.4
│  │
│  ├─ Exported Binaries
│  │  └─ acorn
│  │
│  └─ Dependents
│     └─ jsdom@npm:11.5.1
│
├─ acorn@npm:6.4.2
│  ├─ Version: 6.4.2
│  │
│  ├─ Exported Binaries
│  │  └─ acorn
│  │
│  └─ Dependents
│     ├─ @gulp-sourcemaps/identity-map@npm:2.0.1
│     ├─ acorn-globals@npm:4.3.4
│     ├─ gulp-sourcemaps@npm:3.0.0
│     ├─ gulp-sourcemaps@patch:gulp-sourcemaps@npm%3A3.0.0#./.yarn/patches/gulp-sourcemaps-npm-3.0.0-1ae0fbef6d.patch::version=3.0.0&hash=352196&locator=metamask-crx%40workspace%3A.
│     ├─ webpack@npm:4.46.0
│     └─ webpack@npm:4.46.0 [15918]
│
├─ acorn@npm:7.4.1
│  ├─ Version: 7.4.1
│  │
│  ├─ Exported Binaries
│  │  └─ acorn
│  │
│  └─ Dependents
│     ├─ @storybook/react@npm:6.5.10
│     ├─ @storybook/react@npm:6.5.10 [15918]
│     └─ acorn-jsx@npm:5.3.2 [37b7a]
│
├─ acorn@npm:8.8.0
│  ├─ Version: 8.8.0
│  │
│  ├─ Exported Binaries
│  │  └─ acorn
│  │
│  └─ Dependents
│     ├─ acorn-import-assertions@npm:1.8.0 [f6da4]
│     ├─ acorn-jsx@npm:5.3.2 [c70fa]
│     ├─ espree@npm:9.3.1
│     ├─ espree@npm:9.3.2
│     ├─ jsdom@npm:20.0.0
│     ├─ jsdom@npm:20.0.0 [15ca4]
│     ├─ terser@npm:4.8.0
│     ├─ terser@npm:5.15.0
│     ├─ ts-node@npm:10.5.0
│     ├─ ts-node@npm:10.5.0 [15918]
│     ├─ vm2@npm:3.9.11
│     ├─ webpack@npm:5.74.0
│     └─ webpack@npm:5.74.0 [37b7a]
│
└─ acorn@patch:acorn@npm%3A7.4.1#.yarn/patches/acorn-npm-7.4.1-f450b4646c.patch::version=7.4.1&hash=9c9719&locator=metamask-crx%40workspace%3A.
   ├─ Version: 7.4.1
   │
   ├─ Exported Binaries
   │  └─ acorn
   │
   └─ Dependents
      ├─ acorn-globals@npm:6.0.0
      └─ acorn-node@npm:1.8.2
```

</details>